### PR TITLE
YCM README.md jdt.ls docs

### PR DESCRIPTION
**Pull request by bstaletic**
# PR Prelude

Thank you for working on YCM! :)

**Please complete these steps and check these boxes (by putting an `x` inside
the brackets) _before_ filing your PR:**

- [x] I have read and understood YCM's [CONTRIBUTING][cont] document.
- [x] I have read and understood YCM's [CODE_OF_CONDUCT][code] document.
- [x] I have included tests for the changes in my PR. If not, I have included a
  rationale for why I haven't.
- [x] **I understand my PR may be closed if it becomes obvious I didn't
  actually perform all of these steps.**

# Why this change is necessary and useful

Well, someone had to start writing documentation.

[Please explain **in detail** why the changes in this PR are needed.]

[cont]: https://github.com/Valloric/YouCompleteMe/blob/master/CONTRIBUTING.md
[code]: https://github.com/Valloric/YouCompleteMe/blob/master/CODE_OF_CONDUCT.md

---
This change is [Reviewable](https://reviewable.io/reviews/puremourning/youcompleteme/2)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -942,7 +1002,7 @@ necessary for your project. That should be enough for 99% of projects.
 You could also consider using [YCM-Generator][ygen] to generate the
 `ycm_extra_conf.py` file.
 
-#### Errors during compilation
+#### Errors during compilaton
```
Comment (puremourning) [inline README.md]: oops

File: README.md [hunk 2]
```diff
@@ -1742,43 +1830,6 @@ Default: `0`
 let g:ycm_min_num_identifier_candidate_chars = 0
 ```
 
-### The `g:ycm_max_num_candidates` option
```
Comment (puremourning) [inline README.md]: i think either my branch, or your branch is out of date ? there are unexpected diffs ?

**Review by puremourning — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (bstaletic) [reply README.md]: Somehow, my branch was seriously out of date. That's fixed now.

**Comment by puremourning** [comment]
Thanks for doing this, i totally did use it :) 

Closing it now that it's incorporated into the YCM PR proper.

**Pull request closed without merging**